examples/file_server/src/controller.erl

```diff
@@ -0,0 +1,63 @@
+-module(controller).
+
+-export([
+    execute/2,
+	resource_exists/1,
+	content_types_provided/1
+]).
+
+execute(Req, Env) ->
+	{ok, Req, Env#{ cowmachine_controller => ?MODULE }}.
+
+% Controller export
+
+resource_exists(Context) ->
+	Path = cowmachine_req:path(Context),
+	io:format("Path=~p~n",[Path]),
+	
+	ResultContext = case Path of
+		<<"/">> ->
+			IndexHtml = file(code:priv_dir(main), "examples.html"),
+			cowmachine_req:set_resp_body({file, IndexHtml}, Context);
+		<<"/assets/", FileName/binary>>	->
+			FileNameList = binary_to_list(FileName),
+			File = file(filename:join([code:priv_dir(main), "static/assets"]), FileNameList),
+			cowmachine_req:set_resp_body({file, File}, Context);	
+		_ ->
+			[_|FileName] = binary_to_list(Path),
+			File = file(code:priv_dir(main), FileName),
+			% io:format("File=~p~n",[File]),
+			cowmachine_req:set_resp_body({file, File}, Context)	
+	end,
+	{true, ResultContext}.
+
+content_types_provided(Context) ->
+	{ContentType, NewContext} =  setContentType(Context),
+	{[ContentType], NewContext}.	
+
+%% internal functions
+
+file(DirName, FileName) ->
+	{ok, CurrentDir} = file:get_cwd(),
+	ParentDir = filename:join([CurrentDir, DirName]),
+	filename:join([ParentDir, FileName]).

```

Comment:
Maybe we need to add here something like `mochiweb_util:safe_relative_path/1` - just as an example that we need to be safe.

Reply:
Maybe. It is necessary be safe (checking for the existence of a file) in any way.
In the developed examples, I offer several options (types of projects). Each of them has its pros and cons for that point.

Reply:
Or we just add a comment here to warn the user to sanitize the path before concatenation the file name to the directory. Then it is up to the programmer taking inspiration from this code.

Reply:
I think the comment about it would be enough. [filename:join](https://www.erlang.org/doc/man/filename.html#join-1) make a lot of sanitizing I think.